h5py/tests/test_dtype.py

```diff
@@ -47,7 +47,7 @@ def a(items):
 
         dt_vb = np.dtype([
             ('foo', vidt),
-            ('logical', np.bool)])
```

Comment:
I *think* we want to use the numpy boolean dtype (rather than the python boolean type) here, not sure what it's named though.

Reply:
(It would be the same for the rest of the file)

Reply:
One can use np.bool_ and np.int_ as stated above.

Reply:
From what I can see [in the numpy docs](https://numpy.org/doc/stable/reference/arrays.dtypes.html#specifying-and-constructing-data-types), specifying `bool` for a dtype is equivalent to `np.bool_` anyway. So I think this is fine as is.